integrations/weights_bias/src/haystack_integrations/tracing/weave/tracer.py

```diff
@@ -0,0 +1,222 @@
+        """
+
+        content_tracing_enabled = os.getenv("HAYSTACK_CONTENT_TRACING_ENABLED", "false").lower()
+        haystack_tracer.is_content_tracing_enabled = content_tracing_enabled == "true"
+        if not haystack_tracer.is_content_tracing_enabled:
+            logger.warning(
+                "Traces will not be logged to Weave because Haystack tracing is disabled. "
+                "To enable, set the HAYSTACK_CONTENT_TRACING_ENABLED environment variable to true "
+                "before importing Haystack."
+            )
+
+        self._client = weave.init(project_name)
+        self._current_span: Optional[WeaveSpan] = None
+
+    @staticmethod
+    def create_call(
+        attributes: dict, client: WeaveClient, parent_span: Union[WeaveSpan, None], operation_name: str
+    ) -> Call:
+        comp_name = attributes.pop("haystack.component.name", "")
+        comp_type = attributes.pop("haystack.component.type", "")
+        comp_input = attributes.pop("haystack.component.input", {})
+        call = client.create_call(
+            op=operation_name,
+            inputs={key: coerce_tag_value(value) for key, value in comp_input.items()},
+            attributes={key: coerce_tag_value(value) for key, value in attributes.items()},
+            display_name=f"{comp_name}[{comp_type}].run",
+            parent=parent_span.raw_span() if parent_span else None,
+        )
+        return call
+
+    def current_span(self) -> Optional[Span]:
+        """Get the current active span."""
+        return self._current_span
+
+    @staticmethod
+    def _create_component_span(parent_span: Optional[WeaveSpan], operation_name: str) -> WeaveSpan:
+        """Create a span for component runs with deferred call creation."""
+        return WeaveSpan(parent=parent_span.raw_span() if parent_span else None, operation=operation_name)
+
+    def _create_regular_span(
+        self, operation_name: str, tags: Optional[dict], parent_span: Optional[WeaveSpan]
+    ) -> WeaveSpan:
+        """Create a span for regular operations with immediate call creation."""
+        call = self._client.create_call(
+            op=operation_name,
+            inputs=tags,
+            parent=parent_span.raw_span() if parent_span else None,
+        )
+        return WeaveSpan(call=call)
+
+    def _finish_component_call(self, span: WeaveSpan, parent_span: Optional[WeaveSpan], operation_name: str) -> None:
+        """Create and finish call for component runs."""
+        attributes = span.get_attributes()
+        call = self.create_call(
+            attributes=attributes,
+            client=self._client,
+            parent_span=parent_span,
+            operation_name=operation_name,
+        )
+        span.set_call(call)
+        comp_output = attributes.pop("haystack.component.output", {})
+        self._client.finish_call(
+            call,
+            output={key: coerce_tag_value(value) for key, value in comp_output.items()},
```

Comment:
We could move the coerce_tag_value to `set_tag` so that we don't need to do it in two places (for inputs and outputs) but not absolutely necessary for me.

Reply:
I'm not sure how set tag is supposed to work, and if we need to call `coerce_tag_value` for all input in `set_Tag`- I would rather leave it as is for now - we can change it later.